controllers/provider-aws/pkg/controlplane/actuator.go

```diff
@@ -0,0 +1,359 @@
+	}
+
+	// Create chart applier
+	a.chartApplier, err = gardenerkubernetes.NewChartApplierForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, "could not create chart applier")
+	}
+
+	return nil
+}
+
+// InjectClient injects the given client into the actuator.
+func (a *actuator) InjectClient(client client.Client) error {
+	a.client = client
+	return nil
+}
+
+// Reconcile reconciles the given controlplane and cluster, creating or updating the additional Shoot
+// control plane components as needed.
+func (a *actuator) Reconcile(
+	ctx context.Context,
+	cp *extensionsv1alpha1.ControlPlane,
+	cluster *extensionscontroller.Cluster,
+) error {
+	// Decode providerConfig
+	cpConfig := &apisaws.ControlPlaneConfig{}
+	if _, _, err := a.decoder.Decode(cp.Spec.ProviderConfig.Raw, nil, cpConfig); err != nil {
+		return errors.Wrapf(err, "could not decode providerConfig of controlplane '%s'", objectName(cp))
+	}
+
+	// Decode infrastructureProviderStatus
+	infraStatus := &apisaws.InfrastructureStatus{}
+	if _, _, err := a.decoder.Decode(cp.Spec.InfrastructureProviderStatus.Raw, nil, infraStatus); err != nil {
+		return errors.Wrapf(err, "could not decode infrastructureProviderStatus of controlplane '%s'", objectName(cp))
+	}
+
+	// Deploy secrets
+	a.logger.Info("Deploying secrets", "controlplane", objectName(cp))
+	deployedSecrets, err := controlPlaneSecrets.Deploy(a.cs, a.gcs, cp.Namespace)
+	if err != nil {
+		return errors.Wrapf(err, "could not deploy secrets for controlplane '%s'", objectName(cp))
+	}
+
+	// Determine subnet ID and zone
+	var subnetID, zone string
+	for _, subnet := range infraStatus.VPC.Subnets {
+		if subnet.Purpose == apisaws.PurposePublic {
+			subnetID = subnet.ID
+			zone = subnet.Zone
+			break
+		}
+	}
+	if subnetID == "" || zone == "" {
+		return errors.Errorf("could not determine subnet ID or zone from infrastructureProviderStatus of controlplane '%s'", objectName(cp))
+	}
+
+	// Collect additional configuration chart values
+	cloudProviderConfig := fmt.Sprintf(
+		`[Global]
+VPC=%q
+SubnetID=%q
+DisableSecurityGroupIngress=true
+KubernetesClusterTag=%q
+KubernetesClusterID=%q
+Zone=%q
+`,
+		infraStatus.VPC.ID, subnetID, cp.Namespace, cp.Namespace, zone)
+	values := map[string]interface{}{
+		"cloudProviderConfig": cloudProviderConfig,
+	}
+
+	// Apply config chart
+	a.logger.Info("Applying configuration chart", "controlplane", objectName(cp), "chart", configChart.Name, "values", values)
+	err = configChart.Apply(ctx, a.gcs, a.chartApplier, cp.Namespace, cluster.Shoot, nil, nil, values)
+	if err != nil {
+		return errors.Wrapf(err, "could not apply configuration chart for controlplane '%s'", objectName(cp))
+	}
+
+	// Get cloudprovider secret and configmap from cluster
+	cpSecret, err := a.getSecret(cp.Namespace, common.CloudProviderSecretName)
+	if err != nil {
+		return err
+	}
+	senConfigMap, err := a.getConfigMap(cp.Namespace, common.CloudProviderConfigName)
```

Comment:
```suggestion
	awsConfigMap, err := a.getConfigMap(cp.Namespace, common.CloudProviderConfigName)
```

Reply:
Thanks, it should actually be `cpConfigMap` to be consistent with the rest.

Reply:
Now renamed.